server/chat-plugins/info.js

```diff
@@ -2380,6 +2386,56 @@ const commands = {
 	],
 };
 
+/** @type {PageTable} */
+const pages = {
+	punishments(query, user, connection) {
+		this.title = 'Punishments';
+		let buf = "";
+		this.extractRoom();
+		if (!user.named) return Rooms.RETRY_AFTER_LOGIN;
+		buf += `<div class="pad"><h2>List of active punishments:</h2>`;
+		if (!user.can('mute')) {
+			return buf + `<div class="notice message-error">Access denied.</div>`;
+		}
+		const store = new Map();
+		const muteQueue = this.room.muteQueue;
+		const possessive = (word) => {
+			const suffix = word.endsWith('s') ? `'` : `'s`;
+			return `${word}${suffix}`;
+		};
+
+		for (let [key, value] of Punishments.roomUserids.get(this.room.id)) {
+			if (!store.has(value)) store.set(value, [new Set([value.id]), new Set()]);
+			store.get(value)[0].add(key);
+		}
+
+		for (let [key, value] of Punishments.roomIps.get(this.room.id)) {
+			if (!store.has(value)) store.set(value, [new Set([value.id]), new Set()]);
+			store.get(value)[1].add(key);
+		}
+
+		for (const punishment of store) {
+			let [[punishType, id, expireTime]] = punishment;
+			console.log("Punishment", punishment);
+			let expiresIn = new Date(expireTime).getTime() - Date.now();
+			let expireString = Chat.toDurationString(expiresIn, {precision: 1});
+			buf += `<p>- ${possessive(id)} ${punishType.toLowerCase()} expires in ${expireString}.</p>`;
```

Comment:
Also I told you multiple times already, you're throwing away the names and ips of the users who are actually punished. :/

You should add the names and IPs from the Sets here.